app/Http/Filters/Filter.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Http\Filters;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request; 
+
+abstract class Filter
+{
+    /**
+     * The request instance.
+     *
+     * @var \Illuminate\Http\Request
+     */
+    protected $request;
+
+    /**
+     * The builder instance.
+     *
+     * @var \Illuminate\Database\Eloquent\Builder
+     */
+    protected $builder;
+
+    /**
+     * Initialize a new filter instance.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return void
+     */
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+    }
+
+    /**
+     * Apply the filters on the builder.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $builder
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function apply(Builder $builder): Builder
+    {
+        $this->builder = $builder;
+
+        foreach ($this->request->all() as $name => $value) {
+            if (method_exists($this, $name)) {
```

Comment:
A lot of dirty php approach, that I recommend to avoid. Especially as it is not part of this functionality.